src/types/Check.ts

```diff
@@ -9,40 +9,43 @@ export interface CheckWarrant {
     object: WarrantObject | WarrantObjectLiteral;
     relation: string;
     subject: WarrantObject | Subject;
-    context?: Context;
+    context?: PolicyContext;
 }
 
 export default interface Check extends CheckWarrant {
-    consistentRead?: boolean;
     debug?: boolean;
 }
 
 export interface CheckMany {
     op?: CheckOp;
     warrants: CheckWarrant[];
-    consistentRead?: boolean;
     debug?: boolean;
 }
 
 export interface FeatureCheck {
     featureId: string;
     subject: WarrantObject | Subject;
-    context?: Context;
-    consistentRead?: boolean;
+    context?: PolicyContext;
     debug?: boolean;
 }
 
 export interface PermissionCheck {
     permissionId: string;
     subject: WarrantObject | Subject;
-    context?: Context;
-    consistentRead?: boolean;
+    context?: PolicyContext;
     debug?: boolean;
 }
 
+export interface CheckWarrantRequest {
```

Comment:
Can we just call this `WarrantCheck` or `CheckWarrant`? `CheckWarrantRequest` is kind of confusing since we also have `AccessCheckRequest`

Reply:
We have a `CheckWarrant` defined already; we could add a `WarrantCheck` on top of that although that feels just as confusing. `WarrantCheck` would be the same as `CheckWarrant` but with `object` split up into `objectType` and `objectId`.

If we want to avoid any new naming, then I can keep it with the original change, but update the `authorize` method to split up the `object` field before the request